src/utils/extrinsic-handler.ts

```diff
@@ -0,0 +1,83 @@
+import { SubmittableExtrinsic, ApiTypes, AugmentedEvent } from "@polkadot/api/types";
+import { ISubmittableResult } from "@polkadot/types/types";
+import { newMonetaryAmount, tokenSymbolToCurrency } from "./currency";
+import { TransactionAPI } from "../parachain";
+import { Currency, MonetaryAmount } from "@interlay/monetary-js";
+/**
+ * This class exposes extrinsic metadata and allows simple extrinsic submission,
+ * fee estimation or batching with other extrinsics.
+ */
+class ExtrinsicHandler {
+    constructor(
+        private transactionAPI: TransactionAPI,
+        public extrinsic: SubmittableExtrinsic<"promise">,
+        public event?: AugmentedEvent<"promise">
+    ) {}
+
+    /**
+     * Getter for fee estimation of the extrinsic.
+     *
+     * @returns {MonetaryAmount<Currency>} amount of native currency that will be paid as transaction fee.
+     * @note This fee estimation does not include tip.
+     */
+    public async getFeeEstimation(): Promise<MonetaryAmount<Currency>> {
+        const nativeCurrency = tokenSymbolToCurrency(
+            this.transactionAPI.api.consts.currency.getNativeCurrencyId.asToken
+        );
+
+        const account = this.transactionAPI.getAccount();
+        if (account === undefined) {
+            return newMonetaryAmount(0, nativeCurrency);
+        }
+
+        const paymentInfo = await this.extrinsic.paymentInfo(account);
+        return newMonetaryAmount(paymentInfo.partialFee.toString(), nativeCurrency);
+    }
+
+    /**
+     * This method submits the extrinsic signed by account set in TransactionAPI. Wrapper around TransactionAPI.sendLogged.
+     *
+     * @param {boolean} onlyInBlock Optional parameter to set waiting for transaction inclusion only (true) or finalisation (false).
+     *                              Defaults to true.
+     * @returns {ISubmittableResult} Result of the extrinsic submission.
+     */
+    public submit(onlyInBlock: boolean = true): Promise<ISubmittableResult> {
```

Comment:
I wonder if an enum could be more readable here than the boolean?

Reply:
Yes, possibly. How would you name the enum to make it more readable?)

Reply:
> Yes, possibly. How would you name the enum to make it more readable?)

Maybe a shortened ExtrinsicStatus with only `isInBlock` and `isFinalized` as those are the two options that `DefaultTransactionsApi` looks out for?

(See the full [ExtrinsicStatus definition](https://github.com/polkadot-js/api/blob/4fd3289739e2d0b14eddf7590f729f17f49477cc/packages/types/src/interfaces/author/types.ts#L17-L35) here)

Reply:
I would just make it a pass-through of the `ExtrinsicStatus` and not redefine the states here. The `ExtrinsicStatus` is pretty clearly named and defined and no need to reinvent the wheel on our side.